app/Http/Controllers/TradeOfferController.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\TradeStoreRequest;
+use App\Http\Requests\TradeUpdateRequest;
+use App\Models\Character;
+use App\Models\Item;
+use App\Models\Trade;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+use function PHPUnit\Framework\isNull;
+
+class TradeOfferController extends Controller
+{
+    /**
+     * Retrieve character's items that are valid to trade
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function create(Request $request)
+    {
+        //REQUEST DOES NOT HAVE TRADE_ID NOR CHARACTER_ID . VALIDATION FAILS
+        $offerData = $request->validate([
+            'trade_id' => 'required|exists:trades,id|integer',
+            'character_id' => "required|exists:characters,id|integer",
+        ]);
+
+        $character = Character::where('user_id', Auth::id())
+            ->findOrFail($offerData['character_id']);
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $tradeItem = $trade->item()->get();
+
+        $validItems = $character->items()->where('tier', $tradeItem->tier);
+
+        return view('offer.create', compact('validItems'));
+    }
+
+    /**
+     * Retrieve Trade Offer to edit
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Request $request)
+    {
+        //rrequest params passed in "headers" instead of request parameters...
+
+        $offerData = $request->validate([
+            'trade_id' => 'required|exists:trades,id|integer',
+            'item_id' => 'required|exists:items,id|integer',
+        ]);
+
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $item = Item::findOrFail($offerData['item_id']);
+
+        return view('offer.edit', compact('item', 'trade'));
+    }
+
+    /**
+     *store trade offer
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(Request $request)
+    {
+        $offerData = $request->validate([
+            'item_id' => 'required|exists:items,id|integer',
+            'trade_id' => 'required|exists:trades,id|integer'
+        ]);
+
+        //findOrFail redundant?
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $dataItem = Item::findOrFail($offerData['item_id']);
```

Comment:
you're right, it is redundant, but it doesn't add much overhead, you're call if you want to keep it or not